solv/tdnfpackage.c

```diff
@@ -820,6 +820,58 @@ SolvGetPackageId(
     goto cleanup;
 }
 
+uint32_t
+SolvGetPkgChecksumFromId(
+    PSolvSack pSack,
+    uint32_t dwPkgId,
+    char** ppszChecksum)
+{
+
+    uint32_t dwError = 0;
+    const char* pszTemp = NULL;
+    char* pszChecksum = NULL;
+    Solvable *pSolv = NULL;
+    int checksumtype= REPOKEY_TYPE_SHA256;
+
+    if(!pSack || !pSack->pPool || !ppszChecksum)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    pSolv = pool_id2solvable(pSack->pPool, dwPkgId);
+    if(!pSolv)
+    {
+        dwError = ERROR_TDNF_NO_DATA;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    pszTemp = solvable_lookup_checksum(pSolv, SOLVABLE_CHECKSUM, &checksumtype);
```

Comment:
Looks like `libsolv` has a function to get the checksum in binary: https://github.com/openSUSE/libsolv/blob/2fc0cb31a2010d70f3f9197189730400edbd6ef4/src/solvable.h#L76